crates/rust-analyzer/src/handlers/notification.rs

```diff
@@ -347,8 +365,11 @@ fn run_flycheck(state: &mut GlobalState, vfs_path: VfsPath) -> bool {
                     if id == flycheck.id() {
                         updated = true;
                         match package.filter(|_| !world.config.flycheck_workspace()) {
```

Comment:
```suggestion
                        match package.filter(|_| !world.config.flycheck_workspace() || target_is_bin.is_some()) {
```
As per https://github.com/rust-lang/rust-analyzer/pull/17912#discussion_r1724704318